test/plugin.test.ts

```diff
@@ -0,0 +1,184 @@
+            lastVersionUpdatedPlugin["version"]
+          )
+        ).toBe(true);
+      }
+
+      //  all the other plugins remains the same except the updated one
+      const notUpdatedPlugins = plugins.filter(
+        (plugin, index) =>
+          plugin.version == lastVersion[index].version &&
+          plugin.name == lastVersion[index].name
+      );
+      for (const data of notUpdatedPlugins) {
+        expect(plugins).toContainEqual(data);
+      }
+    }
+  });
+  test("duplicated plugins are not allowed", () => {
+    // filter unique elements in plugins
+    const uniquePlugins = plugins.filter(
+      (plugin, index) => plugins.indexOf(plugin) == index
+    );
+    expect(uniquePlugins).toHaveLength(plugins.length);
+  });
+});
+
+describe("tests on updated or new plugin", () => {
+  // define the list of platforms
+  const platforms = [
+    { name: "windows", pattern: /_windows-amd64.exe$/ },
+    { name: "linux", pattern: /_linux-amd64$/ },
+    { name: "macos-arm64", pattern: /_darwin-arm64$/ },
+    { name: "macos-amd64", pattern: /_darwin-amd64$/ },
+  ];
+
+  let plugins: any[];
+  let lastVersion: any[];
+  let plugin: any;
+  type PluginAsset = {
+    url: string;
+    checksum: string;
+    files?: string[];
+  };
+
+  let pluginAssets: { [key: string]: PluginAsset } = {};
+
+  beforeAll(async () => {
+    plugins = await jsonfile.readFile("plugins.json");
+
+    // Fetch the last version of the plugins file from GitHub
+    lastVersion = await fetch(
+      "https://raw.githubusercontent.com/massalabs/thyra-plugin-store/main/plugins.json"
+    ).then((res) => res.json());
+
+    // Find the plugin that has been updated or added in the new version of the plugins file
+    const newPlugin = plugins.filter(
+      (plugin, index) => plugin.name !== lastVersion[index].name
+    );
+    plugin =
+      newPlugin.length > 0
+        ? newPlugin
+        : plugins.filter(
+            (plugin, index) => plugin.version !== lastVersion[index].version
+          )[0];
+
+    // Extract the assets for the plugin and retrieve the file list and checksum for each asset
+    pluginAssets =
+      plugin &&
+      (await Promise.all(
+        Object.entries(plugin.assets).map(
+          async ([key, asset]: [string, PluginAsset]) => {
+            const files = await getZipFileList(asset.url);
+            const checksum = asset.checksum;
+            return [key, { ...asset, files, checksum }] as [
+              string,
+              PluginAsset
+            ];
+          }
+        )
+      ).then(Object.fromEntries));
+  }, 30000);
+  test("plugin's logo is a valid image", async () => {
+    const dimLogo = await getImageDimensions(plugin.logo);
+    expect(!!dimLogo).not.toBeNull();
+    expect(dimLogo!.width!<40).toBe(true);
+  });
+  platforms.forEach(({ name, pattern }) => {
+    // tests triggered if there is a new or updated plugin
+
+    test(`${name} zip contains a ${name} executable`, async () => {
```

Comment:
I wouldn't use loop in tests. Tests allow people to understand what is happening. It's a way to learn the code. If we add more logic we could be easily lost.

Is there an special reason for that I am missing!?

Otherwise you should cover each case explicitly. It will help to understand what we do in each case. Let me know if I am not clear. slight_smile